src/CppAst/CppNoneTypeTemplateParameterType.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Alexandre Mutel. All rights reserved.
+// Licensed under the BSD-Clause 2 license.
+// See license.txt file in the project root for full license information.
+
+using System;
+
+namespace CppAst
+{
+    /// <summary>
+    /// A C++ template parameter type.
+    /// </summary>
+    public sealed class CppNoneTypeTemplateParameterType : CppType
+    {
+        /// <summary>
+        /// Constructor of this none type template parameter type.
+        /// </summary>
+        /// <param name="name"></param>
+        public CppNoneTypeTemplateParameterType(string name, CppType none_template_type) : base(CppTypeKind.NoneTypeTemplateParameterType)
```

Comment:
`templateNonType` instead of `none_template_type`, also no underscores for C# parameter names.